src/lekcja3_zadanie4_ferta_specjalna_i_diagram/SpecialOffer.java

```diff
@@ -0,0 +1,25 @@
+package lekcja3_zadanie4_ferta_specjalna_i_diagram;
+
+public class SpecialOffer {
+
+    String description;
+    String durationFromTo;
+    double discount;
+    Product product;
+
+    SpecialOffer(String de, String du, double di, Product p) {
+
+        description = de;
+        durationFromTo = du;
+        discount = di;
+        product = p;
+    }
+
+    void discountValue() {
+        product.price = product.price * (1 - discount);
```

Comment:
nie powinnismy modyfikowac ceny bezposrednio w produkcie.( np. co gdybysmy chcieli miec wiele znizek dla tego samego produktu w roznym czasie wtedy wszystskie znizki by sie zsumowały) Lepiej poprostu obliczyć ile wynosi cena po znizce i wyswietlic to wartosc.

Reply:
Poprawione